Enhance specification (#170)
jakartaee/nosql: spec/src/main/asciidoc/chapters/api/entity.adoc

NOTE: Many key-value, wide-column, and document databases feature native support for arrays or even associative arrays of these basic types.

==== Embedded Fields and Embeddable Classes [[embeddable_definition]]

An _embeddable class_ differs from an entity class in that:

- the embeddable class lacks its own persistent identity, and
- the state of an instance of the embeddable class can only be stored in the database when the instance is referenced directly or indirectly by a "parent" entity class instance.

An _embedded field_ is a field whose type is an embeddable class.

Like entities, embeddable classes may have basic fields, embeddable fields, and association fields, but, unlike entities, they do not have identifier fields.

Like entities, a programming model for entity classes might support mutable embeddable classes, immutable embeddable classes, or both.

> A programming model for entity classes might define an annotation that identifies a user-written class as an embeddable class. For example, Jakarta Persistence defines the annotation `jakarta.persistence.Embeddabe`. Alternatively, the programming model might define an annotation that identifies a field as an embedded field. For example, Jakarta Persistence defines the annotation `jakarta.persistence.Embedded`.

Reviewer: ```suggestion
Jakarta NoSQL defines an annotation that identifies a user-written class as an embeddable class: `jakarta.nosql.Embeddable`.
```
